### PR TITLE
Feature/uppsf 4475

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**
# Description

## What

Migrate the service to v2 of the ontology lib.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-4475

## Anything, in particular, you'd like to highlight to reviewers

No business logic should be changed.
Special care was taken when modifying the tests to ensure that is not breaking something. 
See the comment on every test change.

A change to how we generate change logs for concepts update was required to make the tests more stable.

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

## DoD - Ensure all relevant tasks are completed before marking this PR as "Ready for review"

- [ ] Test coverage is not significantly decreased
- [ ] All PR checks have passed
- [ ] Changes are deployed on dev before asking for review
- [ ] Documentations remains up-to-date
  - [ ] OpenAPI definition file is updated
  - [ ] README file is updated
  - [ ] Documentation is updated in upp-docs and upp-public-docs
  - [ ] Architecture diagrams are updated

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

**Comment by angelraynovft** [comment]
chore: fix linter errors https://app.circleci.com/pipelines/github/Financial-Times/concepts-rw-neo4j/430/workflows/2b7de173-08ab-478e-ab32-ce021a2c5157/jobs/1541/parallel-runs/0/steps/0-111

**Comment by ivan-p-nikolov** [comment]
> chore: fix linter errors https://app.circleci.com/pipelines/github/Financial-Times/concepts-rw-neo4j/430/workflows/2b7de173-08ab-478e-ab32-ce021a2c5157/jobs/1541/parallel-runs/0/steps/0-111

done

File: concepts/concepts_service.go [hunk 1]
```diff
@@ -784,13 +792,37 @@ func (s *ConceptService) generateConceptChangeLog(ec ontology.NewAggregatedConce
 	return annotationsChange, string(result), nil
 }
 
-func sortSourceRepresentations(ec ontology.NewAggregatedConcept, nc ontology.NewAggregatedConcept) {
-	sourceMap := make(map[string]ontology.NewConcept)
+func sortConcept(concept ontology.CanonicalConcept) {
```
Comment (epavlova) [inline concepts/concepts_service.go:795]: **thought (non-blocking):** We may find it useful in the future to have this sorting functionality as part of the ontology. It's just a thought.

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module github.com/Financial-Times/concepts-rw-neo4j
 go 1.19
 
 require (
-	github.com/Financial-Times/cm-graph-ontology/v2 v2.0.0-20230824071803-df4140cbc32e
+	github.com/Financial-Times/cm-graph-ontology/v2 v2.0.0-20230927071109-b646623b81f2
```
Comment (epavlova) [inline go.mod:6]: **nitpick:** Just a note so we don't forget this one.

**Review by epavlova — DISMISSED** [review]
🚀

File: go.mod (hunk 2 above)
Comment (ivan-p-nikolov) [reply go.mod]: Fixed.

**Review by epavlova — DISMISSED** [review]

**Review by angelraynovft — DISMISSED** [review]

**Review by angelraynovft — APPROVED** [review]

**Pull request merged**